src/app/components/navigation/navigation.component.ts

```diff
@@ -13,11 +13,16 @@ export class NavigationComponent implements OnInit {
   public isShrunk = false;
   public activeBackToTop = false;
   public lang = 'en';
+  private lang_default = 'en';
```

Comment:
@pnax, if `private lang_default` should always be equal to `public lang` can't you just make them equal in the code? Or is there some use case for them being different?

Reply:
Well, the current code logic need 2 distinct variables because `this.lang` will be overwritten by the current browser language at [line 20 (b35695e)](https://github.com/zonemaster/zonemaster-gui/blob/b35695e3d5c688052f3154ee953d7a3622765a57/src/app/components/navigation/navigation.component.ts#L20) and I needed to keep track of the default language somewhere to set it as a fallback [line 24 (b35695e)](https://github.com/zonemaster/zonemaster-gui/blob/b35695e3d5c688052f3154ee953d7a3622765a57/src/app/components/navigation/navigation.component.ts#L24).

Looking at [line 19 (b35695e)](https://github.com/zonemaster/zonemaster-gui/blob/b35695e3d5c688052f3154ee953d7a3622765a57/src/app/components/navigation/navigation.component.ts#L19) and the call to [`setDefaultLang`](https://github.com/ngx-translate/core#methods), we can think that all this logic is not needed. But somehow the default language is not used as a fallback if the browser language does not match any supported languages. This is incidentally the issue that was raised. To be honest I don't know much of Angular nor NGX-translate, so I coded the logic and kept the current code in case. Maybe the `setDefaultLang` call can be removed.